### PR TITLE
Jenkins testing updates

### PR DESCRIPTION
**Pull request by gardner48**
* Update Jenkins tests to use TPLs built with Spack v0.18.1 and GCC 5.5. 
* Includes fixes from #200

### CODE REVIEW

File: examples/arkode/F2003_serial/ark_kpr_mri_f2003.f90 [hunk 1]
```diff
@@ -571,7 +571,7 @@ program main
   type(c_ptr) :: arkode_mem                ! ARKODE memory structure
   type(c_ptr) :: inner_arkode_mem          ! ARKODE memory structure
   type(c_ptr) :: inner_stepper             ! inner stepper
-  type(c_ptr) :: BTF, BTs                  ! fast/slow method Butcher table
+  type(c_ptr) :: BTf, BTs                  ! fast/slow method Butcher table
```
Comment (drreynolds) [inline examples/arkode/F2003_serial/ark_kpr_mri_f2003.f90:574]: Does this PR depend on #200 (or vice-versa)?  If so, should we merge one before reviewing the other?

Comment (gardner48) [reply examples/arkode/F2003_serial/ark_kpr_mri_f2003.f90:574]: This PR depends on #200 to fix the memory leaks so the Jenkins tests will pass. Merging #200 would make reviewing this PR easier.

File: test/env/default.sh [hunk 2]
```diff
@@ -436,12 +419,15 @@ fi
 
 if [ "$SUNDIALS_PRECISION" == "double" ]; then
     export SUNDIALS_RAJA=ON
-    export RAJA_ROOT="$(spack location -i raja@0.14.0 % "$compiler")"
+    export RAJA_ROOT="$(spack location -i raja@2022.03.1 ~openmp +cuda %"$compiler")"
     export RAJA_BACKENDS="CUDA"
+    # RAJA does not find camp on its own?
+    export camp_ROOT="$(spack location -i camp@2022.03.2 +cuda %"$compiler")"
```
Comment (drreynolds) [inline test/env/default.sh:425]: Thanks!  I kept running into this camp issue when compiling SUNDIALS with RAJA support in the past.

File: cmake/SUNDIALSConfig.cmake.in [hunk 3]
```diff
@@ -146,6 +146,9 @@ if("@ENABLE_SUPERLUMT@" AND NOT TARGET SUNDIALS::SUPERLUMT)
 endif()
 
 if("@ENABLE_RAJA@" AND NOT TARGET RAJA)
+  if(NOT TARGET camp AND NOT camp_DIR)
+    set(camp_DIR "@camp_DIR@")
```
Comment (drreynolds) [inline cmake/SUNDIALSConfig.cmake.in:150]: Thanks! I kept running into this camp issue when compiling SUNDIALS with RAJA support in the past.

**Review by drreynolds — APPROVED** [review]

**Review by balos1 — APPROVED** [review]

**Pull request merged**